heroku/react-hk-components
# Upgrade to React 18

## Why the change?
[GUS Ticket](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE00001IfAYjYAN/view)

## How to verify the changes?
+ visit the PR app 
+ click around and make sure that everything works like in production 
+ tests pass 


## NOTES
+ One of our main dependencies storybook is still using the deprecated method ReactDOM.render in this project. As a result, warnings show up in the console telling you that your app is running as if it was using React 17. From my understanding, we are only using storybook to display our components not create them. Thus, for all intents and purposes react-hk-components itself is on React 18, while some of its dependencies might not be. Storybook has just started adding support for React 18, so in the future, we will probably want to upgrade to allow us to be on the latest version. 

+ I did not update the @types/react, @types/react-dom, @types/react-test-renderer, @types/jest as part of this PR, because because it seems there are a lot of testing issues related to react-popper that cannot be easily fixed. 

+ I also had to add a resolution for chokidar in the package.json because it was causing my local build to fail due to the fsevents version it was using.
